### PR TITLE
Issue #8426: Strip https/http and WWW prefixes when comparing top sites for duplicates

### LINKED ISSUE #8426: Strip https/http and WWW prefixes when comparing top sites for duplicates
In https://github.com/mozilla-mobile/fenix/issues/14442, we can end up showing both kohls as a pinned and frecent site because we are comparing between http://kohls.com and http://www.kohls.com.

We should strip the https/http and WWW prefixes when comparing top sites for duplicates.

### PR DESCRIPTION
**Pull request by gabrielluong**
---

Fixes #8426

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/top-sites/src/test/java/mozilla/components/feature/top/sites/ext/TopSiteKtTest.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        assertTrue(topSites.hasUrl("https://mozilla.com"))
+        assertTrue(topSites.hasUrl("https://www.mozilla.com"))
+        assertTrue(topSites.hasUrl("http://mozilla.com"))
+        assertTrue(topSites.hasUrl("http://www.mozilla.com"))
+        assertTrue(topSites.hasUrl("mozilla.com"))
+
+        assertFalse(topSites.hasUrl("https://mozilla.com/"))
+        assertFalse(topSites.hasUrl("https://m.mozilla.com"))
+        assertFalse(topSites.hasUrl("https://mozilla.com/path"))
+        assertFalse(topSites.hasUrl("https://firefox.com"))
```
Comment (jonalmeida) [inline components/feature/top-sites/src/test/java/mozilla/components/feature/top/sites/ext/TopSiteKtTest.kt:35]: I may also suggest adding some more tests for ensuring the validity as your implementation deps might change which may not be caught:

```suggestion
        assertFalse(topSites.hasUrl("https://firefox.com"))
        assertFalse(topSites.hasUrl("https://mozilla.com/path/is/long"))
        assertFalse(topSites.hasUrl("https://mozilla.com/path#anchor"))
```

**Review by jonalmeida — APPROVED** [review]
This looks good to me.

Although, I think your tests will fail with https://github.com/mozilla-mobile/android-components/pull/8394 merged.

File: components/feature/top-sites/src/test/java/mozilla/components/feature/top/sites/ext/TopSiteKtTest.kt (hunk 1 above)
Comment (gabrielluong) [reply components/feature/top-sites/src/test/java/mozilla/components/feature/top/sites/ext/TopSiteKtTest.kt:35]: Thanks, this good stuff

**Comment by gabrielluong** [comment]
bors r=jonalmeida

**Comment by gabrielluong** [comment]
bors r=jonalmeida

**Pull request merged**